packages/ra-ui-materialui/src/field/ReferenceField.spec.tsx

```diff
@@ -612,6 +615,23 @@ describe('<ReferenceField />', () => {
             });
         });
     });
+    describe('Security', () => {
+        it('should render a link only when users have access to the requested action for the referenced resource', async () => {
```

Comment:
You're not actually testing that. The test should check:

1. the default link is show
2. disabling show access turns the default link to edit
3. Disabling both show and edit should disable the default link